specs/gluster_metrics.adoc

```diff
@@ -0,0 +1,242 @@
+        *** SWAP -- collectd plugin
+        *** Disk IOPS -- collectd plugin
+        *** Network
+            **** Bytes sent and received(rx, tx) -- collectd plugin
+            **** Dropped packets -- collectd plugin
+            **** Errors/overruns -- collectd plugin
+    ** Volume level:
+        *** Status -- gluster get-state glusterd odir /var/run file collectd-glusterd-state
+        *** Bricks counter
+            **** Total -- gluster get-state glusterd...
+            **** Status wise
+        *** Pending Heal -- gluster volume heal all statistics
+        *** Rebalance status -- gluster get-state glusterd..
+        *** Number of connections -- gluster volume status all clients --xml
+    ** Cluster level:
+        *** Status -- Custom logic
+        *** Quorum status --
+        *** Number of nodes
+            **** Total -- gluster get-state glusterd...
+            **** Status-wise --
+        *** Number of volumes
+            **** Total -- gluster get-state glusterd...
+            **** Status wise
+        *** Number of bricks
+            **** Total -- gluster get-state glusterd...
+            **** Staus wise
+
+* A new pluggable approach within our gluster metric collectors will be
+  developed such that the metrics collection plugins are loaded and
+  activated dynamically given just a one time configuration specifying
+  cluster_id and graphite host and port addresses.The plugins hooked to
+  gluster metrics provided pluggable framework are executed in concurrent as
+  green threads(being experimented) and hence their atomicity needs to be
+  ensured.
+    * The cluster topology is learnt once per execution cycle and the same
+      knowledge is shared between plugins unless the same can be fetched
+      by using a different means which is unavoidable.
+      ex: gluster volume status clients info --xml should be used to get
+      connections count but that already readily gives cluster topology
+      which can be used directly.
+    Note:
+        * Above approach runs every plugin in the framework at same interval.
+        * Attempts are being made to have a mechanism by which plugins can
+          register to our framework as light-weight, heavy-weight etc.. and
+          accordingly, the plugin execution interval will differ.
+        Note:
+        ** This is of particular importance in case of certain operations where
+           the command execution too frequently can affect the performance.
+
+* The default threshold configuration for the utilization metrics are:
+
+----
+thresholds:
+  node:
+    cpu:
+      Warning: 80
+      Failure: 90
+    mount_point:
+      Warning: 80
+      Failure: 90
+    memory:
+      Warning: 80
+      Failure: 90
+    swap:
+      Warning: 50
+      Failure: 70
+  gluster:
+    cluster_utilization:
+      Warning: 75
+      Failure: 90
+    volume_utilization:
+      Warning: 75
+      Failure: 90
+    brick_utilization:
+      Warning: 75
+      Failure: 90
```

Comment:
@rishubhjain @GowthamShanmugam I think these threshold values you will need to while implementing grafana's alerting.